[DEVDOCS-5945]: [update] Customers, Update API examples and add contextual info (#275)
bigcommerce/docs: docs/api-docs/customers/customers-subscribers-overview.mdx

Store [settings](https://support.bigcommerce.com/s/article/Checkout-Settings#checkout-settings) can be configured to allow a shopper to complete checkout without creating an account. These shoppers are not captured as customers or stored on BigCommerce. If you want to capture guest data, using the [Storefront APIs](/docs/storefront/cart-checkout/guide/rest-management) can help.

## Customer login API

> The [Customer login API](/docs/start/authentication/customer-login) allows for single sign-on. Once a customer has authenticated by logging in to a third-party system (CMS, portal, or app), you can use the Customer Login API to seamlessly log the customer into their BigCommerce customer account.

Reviewer: ```suggestion
The [Customer login API](/docs/start/authentication/customer-login) allows single sign-on. Once a customer has authenticated by logging in to a third-party system (CMS, portal, or app), you can use the Customer login API to seamlessly log the customer into their BigCommerce customer account.

```

## Current customer API

The [Current customer API](/docs/start/authentication/current-customer) allows your application to identify logged-in customers. This identification is useful when you need to show customer-specific information.

## Customers API

The following endpoints are available:

**GraphQL API**
- Customers GraphQL API

**REST API**
- V3 customers REST API
- V2 customers REST API

### V3 customers API

> The [V3 customers API](/docs/rest-management/customers) is comprised of customers, customer address books, [customer attributes](/docs/rest-management/customers/attributes), customer addresses, customer form fields, and other endpoints that help manage the customer object.

Reviewer: ```suggestion
The [V3 customers API](/docs/rest-management/customers) consists of customers, customer address books, [customer attributes](/docs/rest-management/customers/attributes), customer addresses, customer form fields, and other endpoints that help manage the customer object.

```

[Customer addresses](/docs/rest-management/customers/customer-addresses) consist of a customers address and phone number. Customers can have multiple addresses stored against them.

[Customer form field values](/docs/rest-management/customers/customer-form-field-values) are fields on either the customer address or customer sign-up that accept any string data. For example, a sign-up field can have a customer input a wholesaler ID or, for an address field, have them input any special delivery instructions. For more information about creating form fields, see [Account sign up Fields](https://support.bigcommerce.com/s/article/Editing-Form-Fields#account-fields).

You can access and edit the values for the fields on customer and customer address records using the API.

### V2 Customers API

> The [V2 Customers API](/docs/rest-management/customers-v2) is comprised of customers, customer addresses, and customer groups. The customers object is comprised of basic customer information such as customer name and phone number.

Reviewer: ```suggestion
The [V2 customers API](/docs/rest-management/customers-v2) consists of customers, customer addresses, and customer groups. The customer's object consists of basic customer information such as customer name and phone number.

```

[Customer addresses](/docs/rest-management/customers-v2/customer-addresses) consist of a customer's address and phone number. Customers can have multiple addresses stored against them.

> [Customer groups](/docs/rest-management/customers-v2/customer-groups) allow you to organize your customers, give them discounts, and restrict access to specific products or categories. customer groups are not yet available on the V3 Customers API.

Reviewer: ```suggestion
[Customer groups](/docs/rest-management/customers-v2/customer-groups) let you organize your customers, give them discounts, and restrict access to specific products or categories. Customer groups are not yet available on the V3 customers API.

```

### Upsert

You can use the V3 customers API [Upsert customer attribute values](/docs/rest-management/customers/customer-attribute-values#upsert-customer-attribute-values) operation to upsert form field values and customer attributes. Upsert looks for a match to the existing record and makes an update if there is one. If there is no match, it creates a new record.

### Authentication object

The V3 customers API offers two ways to set a customer's password:

- You can set a new password under the `authentication > new_password` object in a `PUT` or `POST` request.
- To prompt a customer to reset their password, set `force_password_reset` to `true` under `authentication > new_password` object in a `PUT` or `POST` request.

[Password confirmation](/docs/rest-management/subscribers#create-a-subscriber) and [validation](/docs/rest-management/customers-v2/customer-passwords#validate-a-password) are still available under the V2 customers API.

## Subscribers API

The subscribers API allows you to manage subscribers who have signed up for the store’s newsletter or have signed up for abandoned cart emails.

> An important field to note is `consents` - this is what determines whether the Subscriber has been signed up for marketing newsletter emails, abandoned cart emails, or both. This can also be empty, meaning they have not subscribed to either (or may have opted out).

Reviewer: ```suggestion
An important field to note is `consents` - this field determines whether the Subscriber signed up for marketing newsletter emails, abandoned cart emails, or both. This field can also be empty, meaning they have not subscribed to either (or may have opted out).

```
